# GDELT Exploration

## Objective

To explore the GDELT Events dataset, understand its structure,
identify available features, and evaluate its potential usefulness
for our research problem.

In [1]:
import pandas as pd
import requests
import matplotlib.pyplot as plt

In [2]:
import requests
import time
import pandas as pd

url = "https://api.gdeltproject.org/api/v2/doc/doc"

params = {
    "query": "India",
    "mode": "ArtList",
    "maxrecords": 10,
    "format": "json"
}

time.sleep(10)

response = requests.get(url, params=params)

print("Status code:", response.status_code)

Status code: 429


In [3]:
print(response.text[:1000])

Please limit requests to one every 5 seconds or contact kalev.leetaru5@gmail.com for larger queries. All high-traffic users should switch to our ngrams dataset: https://blog.gdeltproject.org/using-the-new-web-ngrams-dataset-to-find-relevant-coverage/. For trend analysis, please see our daily newsletter briefings: https://blog.gdeltproject.org/announcing-our-new-daily-todays-trends-on-capitol-hill-todays-media-trends-newsletter-briefings/.




## Experiment 1: Disaster-Related News Retrieval

The objective of this experiment is to evaluate how GDELT can be used as a source of news data for disaster-related research.

Four disaster categories are considered:

- Flood
- Earthquake
- Cyclone
- Wildfire

For each category, news articles related to India will be retrieved and compared based on the number of articles and available metadata.

In [4]:
import requests
import pandas as pd

### Defining the Disaster Queries

The following queries are used to retrieve news articles related to major types of disasters in India.

In [5]:
disaster_queries = {
    "Flood": "flood India",
    "Earthquake": "earthquake India",
    "Cyclone": "cyclone India",
    "Wildfire": "wildfire India"
}

disaster_queries

{'Flood': 'flood India',
 'Earthquake': 'earthquake India',
 'Cyclone': 'cyclone India',
 'Wildfire': 'wildfire India'}

### Retrieving Flood-Related News

The first experiment focuses on flood-related news in India.

The GDELT DOC 2.0 API is queried for recent articles matching the search term `flood India`.

In [9]:
import time

time.sleep(10)

url = "https://api.gdeltproject.org/api/v2/doc/doc"

params = {
    "query": disaster_queries["Flood"],
    "mode": "artlist",
    "format": "json",
    "maxrecords": 20,
    "timespan": "7d"
}

response = requests.get(url, params=params, timeout=30)

print("Status Code:", response.status_code)

Status Code: 429


In [10]:
if response.status_code == 200:
    data = response.json()
    print("Number of articles returned:", len(data.get("articles", [])))
else:
    print("Request failed.")
    print("Status Code:", response.status_code)
    print("Response:", response.text[:500])

Request failed.
Status Code: 429
Response: Please limit requests to one every 5 seconds or contact kalev.leetaru5@gmail.com for larger queries. All high-traffic users should switch to our ngrams dataset: https://blog.gdeltproject.org/using-the-new-web-ngrams-dataset-to-find-relevant-coverage/. For trend analysis, please see our daily newsletter briefings: https://blog.gdeltproject.org/announcing-our-new-daily-todays-trends-on-capitol-hill-todays-media-trends-newsletter-briefings/.




In [11]:
print(response.headers)

{'Date': 'Sat, 19 Sep 2026 06:46:47 GMT', 'Server': 'GDELT Server', 'Keep-Alive': 'timeout=5, max=20', 'Connection': 'Keep-Alive', 'Transfer-Encoding': 'chunked'}


In [12]:
print(response.text[:1000])

Please limit requests to one every 5 seconds or contact kalev.leetaru5@gmail.com for larger queries. All high-traffic users should switch to our ngrams dataset: https://blog.gdeltproject.org/using-the-new-web-ngrams-dataset-to-find-relevant-coverage/. For trend analysis, please see our daily newsletter briefings: https://blog.gdeltproject.org/announcing-our-new-daily-todays-trends-on-capitol-hill-todays-media-trends-newsletter-briefings/.




## Observation: API Rate Limiting

During the experiment, the GDELT DOC 2.0 API returned HTTP status code 429
(Too Many Requests). The server response indicated that requests should be
limited to one request every 5 seconds.

This shows that GDELT applies request-rate limitations to API users.
Therefore, API requests need to be controlled carefully when conducting
programmatic data collection.

In [13]:
print([name for name in globals() if not name.startswith("_")])

['In', 'Out', 'get_ipython', 'exit', 'quit', 'open', 'pd', 'requests', 'plt', 'time', 'url', 'params', 'response', 'disaster_queries']


## Experiment 2: Retrieving and Structuring Disaster News Data

After identifying the API rate limitation, the next step is to retrieve a
small set of disaster-related news articles and convert the returned
data into a structured Pandas DataFrame for analysis.

In [15]:
import time
import requests
import pandas as pd

# Wait before making a new request
time.sleep(10)

url = "https://api.gdeltproject.org/api/v2/doc/doc"

params = {
    "query": "flood India",
    "mode": "artlist",
    "format": "json",
    "maxrecords": 10,
    "timespan": "7d"
}

response = requests.get(url, params=params, timeout=30)

print("Status Code:", response.status_code)

Status Code: 429


### Result of Experiment 2

The attempt to retrieve a new disaster-related dataset using the GDELT
DOC 2.0 API returned HTTP status code 429 (Too Many Requests).

The server message indicated that API requests should be limited to one
request every five seconds. Even after introducing a delay in the Python
request, the API continued to return the rate-limit response.

Therefore, a new disaster dataset could not be retrieved during this
testing session. The result was recorded as a practical limitation of
using the GDELT API.

## Preliminary Findings

GDELT was explored as a potential source of news and media data for
disaster-related research. The GDELT DOC 2.0 API provides search-based
access to news coverage and supports structured outputs such as JSON,
which can be used for programmatic analysis.

During the practical experiment, API requests were successfully sent to
the GDELT server, but subsequent requests returned HTTP 429 (Too Many
Requests). The server indicated that requests should be limited to one
request every five seconds.

Therefore, the experiment demonstrated both the potential usefulness of
GDELT as a news-data source and an important practical limitation:
programmatic data collection must account for API rate limiting.

Further analysis will be performed only after obtaining a successful
dataset or through an alternative GDELT data-access method.

In [16]:
import pandas as pd

file_path = "../data/20260917.export.CSV.zip"

df = pd.read_csv(
    file_path,
    sep="\t",
    header=None,
    compression="zip",
    low_memory=False
)

print("Rows:", df.shape[0])
print("Columns:", df.shape[1])

Rows: 116212
Columns: 58


In [17]:
df.head()

,0,1,2,3,4,5,6,7,8,9,...,48,49,50,51,52,53,54,55,56,57
0,1323420639,20250917,202509,2025,2025.7041,NaN,NaN,NaN,NaN,NaN,...,TX,2,"Texas, United States",US,USTX,31.1060,-97.64750,TX,20260917,https://finance.yahoo.com/markets/stocks/artic...
1,1323420640,20250917,202509,2025,2025.7041,NaN,NaN,NaN,NaN,NaN,...,-2606561,4,"Rothesay, Argyll and Bute, United Kingdom",UK,UKT8,55.8333,-5.06667,-2606561,20260917,https://www.edp24.co.uk/news/national/26557238...
2,1323420641,20250917,202509,2025,2025.7041,BUS,COMPANY,NaN,NaN,NaN,...,NaN,2,"Texas, United States",US,USTX,31.1060,-97.64750,TX,20260917,https://finance.yahoo.com/markets/stocks/artic...
3,1323420642,20250917,202509,2025,2025.7041,COP,POLICE,NaN,NaN,NaN,...,NaN,3,"Nashville, Tennessee, United States",US,USTN,36.1659,-86.78440,1652484,20260917,https://fox17.com/news/local/markese-greer-arr...
4,1323420643,20250917,202509,2025,2025.7041,COP,POLICE,NaN,NaN,NaN,...,295761,3,"Palm Beach County, Florida, United States",US,USFL,26.6503,-80.44980,295761,20260917,https://fox17.com/news/local/markese-greer-arr...


In [18]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 116212 entries, 0 to 116211
Data columns (total 58 columns):
 #   Column  Non-Null Count   Dtype  
---  ------  --------------   -----  
 0   0       116212 non-null  int64  
 1   1       116212 non-null  int64  
 2   2       116212 non-null  int64  
 3   3       116212 non-null  int64  
 4   4       116212 non-null  float64
 5   5       104772 non-null  str    
 6   6       104772 non-null  str    
 7   7       65783 non-null   str    
 8   8       1647 non-null    str    
 9   9       696 non-null     str    
 10  10      1384 non-null    str    
 11  11      231 non-null     str    
 12  12      49352 non-null   str    
 13  13      3161 non-null    str    
 14  14      66 non-null      str    
 15  15      82134 non-null   str    
 16  16      82134 non-null   str    
 17  17      51801 non-null   str    
 18  18      1662 non-null    str    
 19  19      542 non-null     str    
 20  20      1183 non-null    str    
 21  21      195 non-null 

In [19]:
columns = [
    "GLOBALEVENTID",
    "SQLDATE",
    "MonthYear",
    "Year",
    "FractionDate",
    "Actor1Code",
    "Actor1Name",
    "Actor1CountryCode",
    "Actor1KnownGroupCode",
    "Actor1EthnicCode",
    "Actor1Religion1Code",
    "Actor1Religion2Code",
    "Actor1Type1Code",
    "Actor1Type2Code",
    "Actor1Type3Code",
    "Actor2Code",
    "Actor2Name",
    "Actor2CountryCode",
    "Actor2KnownGroupCode",
    "Actor2EthnicCode",
    "Actor2Religion1Code",
    "Actor2Religion2Code",
    "Actor2Type1Code",
    "Actor2Type2Code",
    "Actor2Type3Code",
    "IsRootEvent",
    "EventCode",
    "EventBaseCode",
    "EventRootCode",
    "QuadClass",
    "GoldsteinScale",
    "NumMentions",
    "NumSources",
    "NumArticles",
    "AvgTone",
    "Actor1Geo_Type",
    "Actor1Geo_FullName",
    "Actor1Geo_CountryCode",
    "Actor1Geo_ADM1Code",
    "Actor1Geo_Lat",
    "Actor1Geo_Long",
    "Actor1Geo_FeatureID",
    "Actor2Geo_Type",
    "Actor2Geo_FullName",
    "Actor2Geo_CountryCode",
    "Actor2Geo_ADM1Code",
    "Actor2Geo_Lat",
    "Actor2Geo_Long",
    "Actor2Geo_FeatureID",
    "ActionGeo_Type",
    "ActionGeo_FullName",
    "ActionGeo_CountryCode",
    "ActionGeo_ADM1Code",
    "ActionGeo_Lat",
    "ActionGeo_Long",
    "ActionGeo_FeatureID",
    "DATEADDED",
    "SOURCEURL"
]

df.columns = columns

print(df.shape)
print(df.columns.tolist())

(116212, 58)
['GLOBALEVENTID', 'SQLDATE', 'MonthYear', 'Year', 'FractionDate', 'Actor1Code', 'Actor1Name', 'Actor1CountryCode', 'Actor1KnownGroupCode', 'Actor1EthnicCode', 'Actor1Religion1Code', 'Actor1Religion2Code', 'Actor1Type1Code', 'Actor1Type2Code', 'Actor1Type3Code', 'Actor2Code', 'Actor2Name', 'Actor2CountryCode', 'Actor2KnownGroupCode', 'Actor2EthnicCode', 'Actor2Religion1Code', 'Actor2Religion2Code', 'Actor2Type1Code', 'Actor2Type2Code', 'Actor2Type3Code', 'IsRootEvent', 'EventCode', 'EventBaseCode', 'EventRootCode', 'QuadClass', 'GoldsteinScale', 'NumMentions', 'NumSources', 'NumArticles', 'AvgTone', 'Actor1Geo_Type', 'Actor1Geo_FullName', 'Actor1Geo_CountryCode', 'Actor1Geo_ADM1Code', 'Actor1Geo_Lat', 'Actor1Geo_Long', 'Actor1Geo_FeatureID', 'Actor2Geo_Type', 'Actor2Geo_FullName', 'Actor2Geo_CountryCode', 'Actor2Geo_ADM1Code', 'Actor2Geo_Lat', 'Actor2Geo_Long', 'Actor2Geo_FeatureID', 'ActionGeo_Type', 'ActionGeo_FullName', 'ActionGeo_CountryCode', 'ActionGeo_ADM1Code', 'Act

In [20]:
print("Minimum SQLDATE:", df["SQLDATE"].min())
print("Maximum SQLDATE:", df["SQLDATE"].max())

print("\nYears present:")
print(df["Year"].value_counts().sort_index())

Minimum SQLDATE: 20160919
Maximum SQLDATE: 20260917

Years present:
Year
2016        16
2025       892
2026    115304
Name: count, dtype: int64


In [21]:
date_counts = df["SQLDATE"].value_counts().sort_index()

print(date_counts.tail(20))

SQLDATE
20160919        16
20250917       892
20260818       432
20260910      1086
20260916       251
20260917    113535
Name: count, dtype: int64


In [22]:
print("Number of unique EventRootCodes:",
      df["EventRootCode"].nunique())

print("\nMost common EventRootCodes:")
print(df["EventRootCode"].value_counts().head(20))

Number of unique EventRootCodes: 20

Most common EventRootCodes:
EventRootCode
4     29429
1     17562
5     10191
2      8586
3      7805
11     6989
19     6113
17     5475
7      3941
8      3823
12     3392
6      2775
9      2655
13     1999
18     1633
10     1566
16     1186
14      681
15      332
20       79
Name: count, dtype: int64


In [23]:
disaster_keywords = [
    "flood",
    "earthquake",
    "cyclone",
    "storm",
    "wildfire",
    "drought",
    "landslide",
    "tsunami"
]

# Search the source URLs for disaster-related keywords
url_text = df["SOURCEURL"].fillna("").str.lower()

for keyword in disaster_keywords:
    count = url_text.str.contains(keyword, regex=False).sum()
    print(f"{keyword}: {count}")

flood: 444
earthquake: 9
cyclone: 0
storm: 239
wildfire: 142
drought: 99
landslide: 9
tsunami: 9


In [24]:
flood_records = df[
    df["SOURCEURL"].fillna("").str.lower().str.contains("flood", regex=False)
].copy()

print("Flood-related URL records:", len(flood_records))

flood_records[
    [
        "SQLDATE",
        "EventCode",
        "EventRootCode",
        "NumMentions",
        "NumSources",
        "NumArticles",
        "AvgTone",
        "ActionGeo_FullName",
        "SOURCEURL"
    ]
].head(10)

Flood-related URL records: 444


,SQLDATE,EventCode,EventRootCode,NumMentions,NumSources,NumArticles,AvgTone,ActionGeo_FullName,SOURCEURL
217,20260917,14,1,16,4,16,-4.908450,"Mosquito Range, Colorado, United States",https://www.postindependent.com/news/western-s...
218,20260917,14,1,32,4,32,-4.908450,"Grand Junction, Colorado, United States",https://www.postindependent.com/news/western-s...
219,20260917,14,1,16,4,16,-4.908450,"Colorado, United States",https://www.postindependent.com/news/western-s...
312,20260917,130,13,4,1,4,-3.248031,"Durban, KwaZulu-Natal, South Africa",https://www.news24.com/southafrica/weather/thu...
313,20260917,130,13,4,1,4,-3.248031,"Kwazulu, Eastern Cape, South Africa",https://www.news24.com/southafrica/weather/thu...
833,20260917,10,1,84,20,84,0.950214,China,http://www.nepalnational.com/news/279311282/ne...
872,20260917,90,9,3,1,3,-4.290429,"Kathmandu, Bagmati, Nepal",http://www.nepalnational.com/news/279311282/ne...
873,20260917,90,9,27,3,27,-2.984601,China,http://www.nepalnational.com/news/279311282/ne...
917,20260917,10,1,1,1,1,-4.290429,"Kathmandu, Bagmati, Nepal",http://www.nepalnational.com/news/279311282/ne...
918,20260917,10,1,1,1,1,-4.290429,China,http://www.nepalnational.com/news/279311282/ne...


In [25]:
# Filtering for India-related flood events
india_records = df[
    (df["ActionGeo_CountryCode"] == "IN") |
    (df["Actor1CountryCode"] == "IND") |
    (df["Actor2CountryCode"] == "IND")
].copy()

print("India-associated records:", len(india_records))
print("Percentage of dataset:", round(len(india_records) / len(df) * 100, 2), "%")

India-associated records: 6952
Percentage of dataset: 5.98 %


In [26]:
disaster_keywords = [
    "flood",
    "earthquake",
    "cyclone",
    "storm",
    "wildfire",
    "drought",
    "landslide",
    "tsunami"
]

india_disaster_records = india_records[
    india_records["SOURCEURL"]
    .fillna("")
    .str.lower()
    .str.contains("|".join(disaster_keywords), regex=True)
].copy()

print("India-associated disaster-related records:",
      len(india_disaster_records))

print("\nDisaster keyword matches:")
for keyword in disaster_keywords:
    count = india_records["SOURCEURL"].fillna("").str.lower().str.contains(
        keyword, regex=False
    ).sum()
    print(f"{keyword}: {count}")

India-associated disaster-related records: 61

Disaster keyword matches:
flood: 34
earthquake: 0
cyclone: 0
storm: 1
wildfire: 0
drought: 26
landslide: 0
tsunami: 0


In [27]:
india_disaster_records[
    [
        "SQLDATE",
        "Actor1Name",
        "Actor2Name",
        "EventCode",
        "EventRootCode",
        "GoldsteinScale",
        "NumMentions",
        "NumSources",
        "NumArticles",
        "AvgTone",
        "ActionGeo_FullName",
        "SOURCEURL"
    ]
].head(20)

,SQLDATE,Actor1Name,Actor2Name,EventCode,EventRootCode,GoldsteinScale,NumMentions,NumSources,NumArticles,AvgTone,ActionGeo_FullName,SOURCEURL
20918,20260917,VILLAGE,DELHI,100,10,-5.0,2,1,2,-3.675676,"Delhi, Delhi, India",https://scroll.in/article/1095410/when-delhi-f...
21142,20260917,LIEUTENANT,VILLAGE,100,10,-5.0,8,1,8,-3.675676,"Delhi, Delhi, India",https://scroll.in/article/1095410/when-delhi-f...
22215,20260917,RUSSIAN,JAIN,43,4,2.8,2,1,2,-1.924353,India,https://asianews.network/no-bodies-no-answers-...
22216,20260917,RUSSIAN,JAIN,43,4,2.8,6,1,6,-1.924353,India,https://asianews.network/no-bodies-no-answers-...
27614,20260917,AMIT,NaN,90,9,-2.0,8,1,8,-2.298851,"New Delhi, Delhi, India",https://www.dailyexcelsior.com/chinta-baithak-...
28504,20260917,CONGRESS,MAHARASHTRA,20,2,3.0,10,1,10,-6.081081,NaN,https://dailypioneer.com/news/congress-leaders...
32450,20260917,MINISTRY,NaN,10,1,0.0,3,1,3,-5.462185,India,https://www.sentinelassam.com/more-news/intern...
43989,20260917,FARMER,NaN,12,1,-0.4,4,1,4,1.206897,"Konkan, Maharashtra, India",https://www.freepressjournal.in/pune/marathwad...
43990,20260917,FARMER,NaN,12,1,-0.4,6,1,6,1.206897,"Pune, Maharashtra, India",https://www.freepressjournal.in/pune/marathwad...
43993,20260917,FARMER,GOVERNMENT,1053,10,-5.0,5,1,5,-7.189542,"Pune, Maharashtra, India",https://www.freepressjournal.in/pune/beed-farm...


In [28]:
print(
    india_disaster_records[
        ["EventCode", "EventRootCode"]
    ].value_counts().head(20)
)

EventCode  EventRootCode
40         4                15
46         4                12
37         3                 8
43         4                 4
70         7                 3
100        10                2
10         1                 2
12         1                 2
60         6                 2
32         3                 2
42         4                 2
90         9                 1
20         2                 1
1053       10                1
110        11                1
36         3                 1
311        3                 1
51         5                 1
Name: count, dtype: int64


In [29]:
pd.set_option("display.max_colwidth", 150)

flood_india = india_records[
    india_records["SOURCEURL"]
    .fillna("")
    .str.lower()
    .str.contains("flood", regex=False)
].copy()

print("Candidate flood records:", len(flood_india))

print(
    flood_india[
        [
            "SQLDATE",
            "EventCode",
            "Actor1Name",
            "Actor2Name",
            "ActionGeo_FullName",
            "NumMentions",
            "SOURCEURL"
        ]
    ].to_string(index=False)
)

Candidate flood records: 34
 SQLDATE  EventCode              Actor1Name              Actor2Name             ActionGeo_FullName  NumMentions                                                                                                                                                          SOURCEURL
20260917        100                 VILLAGE                   DELHI            Delhi, Delhi, India            2                                                                          https://scroll.in/article/1095410/when-delhi-floods-the-villages-around-it-bear-the-brunt
20260917        100              LIEUTENANT                 VILLAGE            Delhi, Delhi, India            8                                                                          https://scroll.in/article/1095410/when-delhi-floods-the-villages-around-it-bear-the-brunt
20260917         43                 RUSSIAN                    JAIN                          India            2                                    

In [30]:
print("Flood candidate event records:", len(flood_india))

print("Unique source articles:", flood_india["SOURCEURL"].nunique())

print("\nArticles by number of event records:")
print(
    flood_india["SOURCEURL"]
    .value_counts()
    .sort_values(ascending=False)
)

Flood candidate event records: 34
Unique source articles: 8

Articles by number of event records:
SOURCEURL
https://aninews.in/news/world/asia/nirmala-sitharaman-meets-nepali-counterpart-swarnim-wagle-reaffirms-indias-support-for-post-flood-reconstruction20260918005856/    17
https://www.thehindu.com/news/national/fm-nirmala-sitharaman-meets-nepalese-counterpart-wagle-discusses-flood-recovery/article71478365.ece                             8
https://scroll.in/article/1095410/when-delhi-floods-the-villages-around-it-bear-the-brunt                                                                              2
https://asianews.network/no-bodies-no-answers-foreign-families-search-for-loved-ones-missing-in-nepal-floods/                                                          2
https://www.indiawaterportal.org/climate-change/disasters/when-floods-wash-away-learning-why-indias-disaster-planning-must-protect-children                            2
https://www.sentinelassam.com/more-news/interna

In [31]:
flood_articles = (
    flood_india[
        [
            "SQLDATE",
            "ActionGeo_FullName",
            "NumMentions",
            "NumSources",
            "NumArticles",
            "AvgTone",
            "SOURCEURL"
        ]
    ]
    .drop_duplicates(subset="SOURCEURL")
    .reset_index(drop=True)
)

print("Unique flood-related source articles:", len(flood_articles))

flood_articles

Unique flood-related source articles: 8


,SQLDATE,ActionGeo_FullName,NumMentions,NumSources,NumArticles,AvgTone,SOURCEURL
0,20260917,"Delhi, Delhi, India",2,1,2,-3.675676,https://scroll.in/article/1095410/when-delhi-floods-the-villages-around-it-bear-the-brunt
1,20260917,India,2,1,2,-1.924353,https://asianews.network/no-bodies-no-answers-foreign-families-search-for-loved-ones-missing-in-nepal-floods/
2,20260917,India,3,1,3,-5.462185,https://www.sentinelassam.com/more-news/international/nepal-forms-7-member-committee-to-determine-origin-cause-sequence-of-events-behind-floods
3,20260917,"Lhonak, Nepal (general), Nepal",6,1,6,-8.281999,https://www.resilience.org/stories/2026-09-17/how-ice-and-rock-triggered-nepals-catastrophic-flood-and-why-climate-change-is-raising-the-risks/
4,20260917,"Delhi, Delhi, India",5,1,5,-1.923077,https://kathmandupost.com/national/2026/09/17/nepal-india-finance-ministers-discuss-post-flood-reconstruction-and-economic-ties
5,20260917,India,10,1,10,-5.907781,https://www.indiawaterportal.org/climate-change/disasters/when-floods-wash-away-learning-why-indias-disaster-planning-must-protect-children
6,20260917,"New Delhi, Delhi, India",2,1,2,1.153846,https://aninews.in/news/world/asia/nirmala-sitharaman-meets-nepali-counterpart-swarnim-wagle-reaffirms-indias-support-for-post-flood-reconstructio...
7,20260917,"New Delhi, Delhi, India",8,1,8,1.030928,https://www.thehindu.com/news/national/fm-nirmala-sitharaman-meets-nepalese-counterpart-wagle-discusses-flood-recovery/article71478365.ece


In [32]:
flood_articles["SOURCEURL"].str.lower().str.contains(
    "nepal",
    regex=False
).value_counts()

SOURCEURL
True     6
False    2
Name: count, dtype: int64

In [33]:
india_flood_articles = flood_articles[
    ~flood_articles["SOURCEURL"]
    .str.lower()
    .str.contains("nepal", regex=False)
].copy()

print("Potential India-focused flood articles:",
      len(india_flood_articles))

india_flood_articles[
    [
        "SQLDATE",
        "ActionGeo_FullName",
        "NumMentions",
        "NumSources",
        "NumArticles",
        "AvgTone",
        "SOURCEURL"
    ]
]

Potential India-focused flood articles: 2


,SQLDATE,ActionGeo_FullName,NumMentions,NumSources,NumArticles,AvgTone,SOURCEURL
0,20260917,"Delhi, Delhi, India",2,1,2,-3.675676,https://scroll.in/article/1095410/when-delhi-floods-the-villages-around-it-bear-the-brunt
5,20260917,India,10,1,10,-5.907781,https://www.indiawaterportal.org/climate-change/disasters/when-floods-wash-away-learning-why-indias-disaster-planning-must-protect-children


In [34]:
print("Potential India-focused flood articles:", len(india_flood_articles))

print("\nSummary statistics:")
print(
    india_flood_articles[
        ["NumMentions", "NumSources", "NumArticles", "AvgTone"]
    ].describe()
)

Potential India-focused flood articles: 2

Summary statistics:
       NumMentions  NumSources  NumArticles   AvgTone
count     2.000000         2.0     2.000000  2.000000
mean      6.000000         1.0     6.000000 -4.791728
std       5.656854         0.0     5.656854  1.578337
min       2.000000         1.0     2.000000 -5.907781
25%       4.000000         1.0     4.000000 -5.349755
50%       6.000000         1.0     6.000000 -4.791728
75%       8.000000         1.0     8.000000 -4.233702
max      10.000000         1.0    10.000000 -3.675676


## Final Findings

The GDELT Event dataset was successfully explored using a downloadable
daily export containing 116,212 event records.

An India-associated subset was created using actor-country and
action-geography fields, resulting in 6,952 records. Disaster-related
candidates were then identified using flood-related keywords in source
URLs.

The initial flood filter produced 34 event records corresponding to
8 unique source articles. This demonstrated that multiple GDELT event
records can originate from the same news article. Therefore, source URLs
were used to obtain an article-level view.

After excluding candidates whose source URLs explicitly referenced
Nepal, 2 potential India-focused flood articles remained.

For these two candidate articles, the average number of mentions was
6, the average number of associated articles was 6, and the average
GDELT tone score was approximately -4.79.

These results should be considered exploratory rather than a validated
India flood dataset. The filtering approach relied on geographic
association and URL keywords, and the GDELT Event dataset does not
contain the full article text. Country association can also arise from
actors or locations mentioned in an article rather than from the
location of the disaster itself.

## Practical Limitations

1. The GDELT DOC API returned HTTP 429 (Too Many Requests) during
   programmatic experimentation, demonstrating the importance of
   rate-limit handling.

2. The downloadable Event dataset provided a practical alternative for
   structured event analysis.

3. Event-level records are not equivalent to individual news articles;
   multiple event records may originate from the same source URL.

4. Simple keyword filtering can produce false positives or geographically
   ambiguous results.

5. The Event dataset provides structured event metadata and source URLs,
   but not the complete article text.

## Research Relevance

GDELT can provide a useful news and event-data modality for a
multimodal disaster-analysis research pipeline. Its structured fields
can support analysis of event frequency, geographic association,
actors, mentions, and media tone.

For a future multimodal system, GDELT data could be combined with
social-media signals and satellite/geospatial observations to study
disasters from complementary perspectives.

This experiment therefore established the practical workflow for
acquiring, structuring, filtering, deduplicating, and critically
evaluating GDELT disaster-related data.